crates/worktree/src/worktree.rs

```diff
@@ -445,10 +432,26 @@ impl Worktree {
             })
             .detach();
 
+            let share_private_files = false;
+            if let Some(metadata) = metadata {
+                let mut entry = Entry::new(
+                    Arc::from(Path::new("")),
+                    &metadata,
+                    &next_entry_id,
+                    snapshot.root_char_bag,
+                    None,
+                );
+                if !metadata.is_dir {
```

Comment:
Note to self, this needs adjusting to deal with symlinks

Reply:
I don't think this is an issue after all; at least I don't see anywhere else that we take symlink resolution into account is the `is_path_private` calculation